src/components/views/Profile.tsx

```diff
@@ -1,33 +1,88 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { useNavigate } from "react-router-dom";
 import { Button } from "components/ui/Button";
 import "styles/views/Login.scss";
 import "styles/views/Profile.scss";
 import BaseContainer from "components/ui/BaseContainer";
 import { format, isValid } from "date-fns";
 import ProfileIcon from "components/ui/ProfileIcon";
+import { api, handleError } from "helpers/api";
+import User from "models/User";
+
 
 const Profile = () => {
     const navigate = useNavigate();
-    const [username, setUsername] = useState("xxGamerxx");
-    const [birthday, setBirthday] = useState<string>("not given");
+    const user = localStorage.getItem("currentUser");
+    localStorage.setItem("test", user);
+    const [username, setUsername] = useState();
+    const [favourite, setFavourite] = useState(null);
     const [isEditing, setIsEditing] = useState(false);
+    const [userData, setUserData] = useState({username: "...", status: "..."});
+
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const userID = localStorage.getItem("userID");
+                let response = await api.get("/users/" + userID);
+                response.data.creationDate = handleDate(response.data.creationDate);
+                const userdata = new User(response.data);
+                console.log(userdata);
+                setUserData(userdata);
+                setUsername(userdata.username);
+
+                if (!userdata.favourite) {
+                    setFavourite("Zaddy");
+                } else {
+                    setFavourite(userdata.favourite);
+                }
+  
+            } catch (error) {
+                alert("Server Connection Lost");
+                navigate("/lobbyoverview");
+            }
+        };
+        fetchData();
+    }, []);
+
+    const updateUserData = async (username, favourite) => {
+        try {
+            const token = localStorage.getItem("token"); 
+            const userID = localStorage.getItem("userID");
+            const config = {
+                headers: {
+                    "Authorization": token,
+                    "Content-Type": "application/json"
+                }
+            };
+    
+            const response = await api.put("/users/" + userID, {
+                username: username,
+                favourite: favourite, 
+            }, config);
+            
+            console.log("User data updated successfully:", response.data);
+        } catch (error) {
+            alert("Failed to update user data");
```

Comment:
This is a rather vague error, please add an error description. (it is useful, for example, when the user tries to change to a username of someone else)